lib/plan-finder.js

```diff
@@ -152,22 +153,50 @@ class PlanFinder {
                 * @param {String} name transformer name
                 * @param {Object} attributes custom attributes for this step
      */
-    formPlan(source, outputInstructions, currentPlan) {
+    async formPlan(source, outputInstructions, currentPlan) {
         if (!currentPlan || currentPlan.length === 0) {
             debug(`No plan could be found for : ${source.type} -> ${outputInstructions.type}`);
             throw new RenditionFormatUnsupportedError(`No plan could be found for : ${source.type} -> ${outputInstructions.type}`);
         }
+
         // userData is an important field for sending IO events in Nui
         // for now, add them to every output step instructions
         // future plan: store these somewhere, maybe in the plan or engine level to put inside every event
         const userData = outputInstructions.userData;
-
         const plan = [];
+
         // first input is always source
-        let input = source;
-        for(let i = 0; i < currentPlan.length; i++) {
+        let input = clone(source);
+
+        // get image metadata and check for "Orientation" property if the asset is going straight to sensei
+        // this means the current plan only has one transformer and it is a sensei transformer
+        if (currentPlan.length === 1 && currentPlan[0].includes("SenseiTransformer")) {
```

Comment:
Do we really want to hardcode specific transformers in here (that are not part of this library)?

I think there should be a generic abstraction for this or other solution.

cc @jdelbick

Reply:
(@alexkli Here we're doing a "quick fix". A more elaborate fix will happen during [this epic](https://jira.corp.adobe.com/browse/NUI-1434), as it deals with improvements and adding support for other things, like explicit orientation handling; and [orientation handling](https://jira.corp.adobe.com/browse/NUI-1543) is here).

Reply:
Ok. If it's a temporary quick fix, I suggest to clearly mark it as such in code comments. And mark the code sections that should be removed later again.